### PR TITLE
Add split selected rows op

### LINKED ISSUE #7558: Split/concat need to support selected rows.

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #7558

### CODE REVIEW

File: paddle/operators/split_selected_rows_op.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+namespace paddle {
+namespace operators {
+
+class SplitSelectedRowsOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  SplitSelectedRowsOpMaker(OpProto *proto, OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X", "The input SelectedRows.");
+    AddOutput("Out", "The outputs of input SelectedRows.").AsDuplicable();
+    AddAttr<std::vector<int>>("rows_section", "Rows section for output.")
```
Comment (typhoonzero) [inline paddle/operators/split_selected_rows_op.cc]: `rows_section` => `rows_sections

File: paddle/operators/split_selected_rows_op.cc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        .SetDefault(std::vector<int>({}));
+    AddAttr<std::vector<int>>("height_section",
```
Comment (typhoonzero) [inline paddle/operators/split_selected_rows_op.cc]: `height_sections`

File: paddle/operators/split_selected_rows_op.cc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                              "Height for each output SelectedRows.")
+        .SetDefault(std::vector<int>({}));
+
+    AddComment(R"DOC(
+Split a SelectedRows with a specified rows section.
+You could set height_section for specified the height for each output.
```
Comment (typhoonzero) [inline paddle/operators/split_selected_rows_op.cc]: `height_sections` is only needed when need to split the dims of the original tensor.

File: paddle/operators/split_selected_rows_op.cc [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            " section size.");
+        dims[0] = height_section[i];
+      }
+      outs_dims.push_back(dims);
+    }
+    ctx->SetOutputsDim("Out", outs_dims);
+  }
+};
+
+class SplitSelectedRowsGradMaker : public framework::SingleGradOpDescMaker {
```
Comment (typhoonzero) [inline paddle/operators/split_selected_rows_op.cc]: I didn't see a grad kernel implementation, did you missed that?

Comment (Yancey1989) [reply paddle/operators/split_selected_rows_op.cc]: We reuse `sum` op as the grad op, and I will add the unit test for it.

Comment (Yancey1989) [reply paddle/operators/split_selected_rows_op.cc]: Done.

File: paddle/operators/split_selected_rows_op.cc (hunk 2 above)
Comment (Yancey1989) [reply paddle/operators/split_selected_rows_op.cc]: Done.

File: paddle/operators/split_selected_rows_op.cc (hunk 3 above)
Comment (Yancey1989) [reply paddle/operators/split_selected_rows_op.cc]: Done.

File: paddle/operators/split_selected_rows_op.cc (hunk 1 above)
Comment (Yancey1989) [reply paddle/operators/split_selected_rows_op.cc]: Done.

File: paddle/operators/split_selected_rows_op.cc (hunk 4 above)
Comment (typhoonzero) [reply paddle/operators/split_selected_rows_op.cc]: Do you need to register the sum kernel as the grad kernel?

Comment (Yancey1989) [reply paddle/operators/split_selected_rows_op.cc]: Maybe don't need, just inherit `framework::SingleGradOpDescMaker `, we can use another Operator as the grad op.

**Pull request merged**